### PR TITLE
fix for https://github.com/py-smart-gardena/hass-gardena-smart-system…

### PR DESCRIPTION
**Pull request by grm**
…/issues/4

### CODE REVIEW

File: src/gardena/smart_system.py [hunk 1]
```diff
@@ -166,9 +164,9 @@ def __call_smart_system_get(self, url):
 
     def update_locations(self):
         response_data = self.__call_smart_system_get(f"{self.SMART_HOST}/v1/locations")
-        if response_data is not None:
-            if len(response_data["data"]) < 1:
-                _LOGGER.error("No locations found....")
+        if response_data is not None or 'data' not in response_data:
```
Comment (osks) [inline src/gardena/smart_system.py:167]: I think you should remove the ´or 'data' not in response_data` from this if case. That is just making sure there wasn't an error. In the next line you're anyway testing if 'data' is in response_data or not.

**Comment by osks** [comment]
How come the Codacy check didn't work?

File: src/gardena/smart_system.py (hunk 1 above)
Comment (grm) [reply src/gardena/smart_system.py:167]: Data may not be in the answer from the api,and I wanted to check that also ... if it is not, len(response_data["data"]) will raise an exception without a message before checking the length :)

**Comment by grm** [comment]
I don t know why codacy did not work. I tried to configure it last night but did not succeed and I was tired after that. Checks are done on master but not on master.

I am try to fix it,but if you find a solution, feel free to do it.

Btw,I think we can ignore that while correcting the bugs and after we ll re enable it

File: src/gardena/smart_system.py (hunk 1 above)
Comment (osks) [reply src/gardena/smart_system.py:167]: If there was an error, response_data is None. (The error is logged and update_locations() can't do anything else with it.) If there wasn't an error, response_data will be a dict.

If there wasn't an error, response_data is a dict, and on the line below (`if 'data' not in response_data or len(response_data["data"]) < 1:`) you check if 'data' is in response_data before using it. That's all that's needed.

Comment (osks) [reply src/gardena/smart_system.py:167]: I've also never seen the case where there wasn't an error and 'data' was not in response_data.

It's fine to check for it (on line 168). But the check on line 167 is hard to reason about, and at the same time there is no need for it as there is a check on line 168.

Comment (grm) [reply src/gardena/smart_system.py:167]: Ok, you probably have a more up-to-date vision of the API return :)

My point was more if there was no data in the response, the line 168 will fail (we are checking the length of 'response_data["data"]' but if data is not in response_data, we will have an exception, but if you did not view the case and it is not mandatory, we should not include it.

I will update the code now ;)

Comment (grm) [reply src/gardena/smart_system.py:167]: ha my bad, i forgot to remove it on line 167 :)

**Comment by grm** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
+ Solved 8
           
```


See the complete overview on [Codacy](https://app.codacy.com/gh/py-smart-gardena/py-smart-gardena/pullRequest?prid=5614304&bid=18392000)

**Pull request merged**